modules/queue/unique_queue_redis.go

```diff
@@ -102,11 +104,7 @@ func (fifo *RedisUniqueByteFIFO) PushFunc(data []byte, fn func() error) error {
 // Pop pops data from the start of the fifo
 func (fifo *RedisUniqueByteFIFO) Pop() ([]byte, error) {
 	data, err := fifo.client.LPop(fifo.queueName).Bytes()
-	if err != nil {
-		return data, err
-	}
-
-	if len(data) == 0 {
+	if err == nil || err == redis.Nil {
```

Comment:
Oops. Not sure how that sneaked in.

Reply:
OK. This is still not right. I sent another commit.